### PR TITLE
Implemented Forgot password Fixes issue#136

### PR DESCRIPTION
**Pull request by sachinprasad1998mav**
@DevangRaval1 @sachinprasad1998mav

### CODE REVIEW

File: frontend/src/Components/ResetPassword/NewPassword.tsx [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+  // const handleSubmit = (event) => {
+  //   event.preventDefault();
+  
+  //   if (npassword !== cnpassword) {
+  //     alert("New password and confirm password do not match.");
+  //     return;
+  //   }
+  
+  // }
+
```
Comment (prasannarajezzzy) [inline frontend/src/Components/ResetPassword/NewPassword.tsx:86]: remove this if not required

File: frontend/src/Components/ResetPassword/NewPassword.tsx [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>): void => {
+    event.preventDefault();
+    console.log("Handle Submit");
+    setOpen(true);
+    // const data = new FormData(event.currentTarget);
+    // const npasswordValue = data.get("npassword") as string;
+    // const cnpasswordValue = data.get("cnpassword") as string;
+    // navigate("/passwordsuccessful")
```
Comment (prasannarajezzzy) [inline frontend/src/Components/ResetPassword/NewPassword.tsx:122]: remove this if not required

File: frontend/src/Components/ResetPassword/NewPassword.tsx [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        aria-describedby="modal-modal-description"
+      >
+        <Box sx={style}>
+          <Typography id="modal-modal-title" variant="h6" component="h2" >
+            Password Changed Successfully!
+          </Typography>
+
+          {/* <Typography id="modal-modal-description" sx={{ mt: 2 }}>
+            Duis mollis, est non commodo luctus, nisi erat porttitor ligula.
+          </Typography> */}
```
Comment (prasannarajezzzy) [inline frontend/src/Components/ResetPassword/NewPassword.tsx:217]: remove this if not required

**Review by mmendan — APPROVED** [review]
Looks good to me.

**Review by murthy-001 — COMMENTED** [review]
LGTM .

**Comment by briswells** [comment]
@sachinprasad1998mav @prasannarajezzzy  @murthy-001 @mmendan Please be careful when creating and reviewing PR's. PantryNodeReach/package-lock.json was committed to main with the PR and it should not be there. That is an outdated structure, and as you can see now we just have a folder that only contains a package-lock.

**Comment by hardikpatil** [comment]
> @sachinprasad1998mav @prasannarajezzzy @murthy-001 @mmendan Please be careful when creating and reviewing PR's. PantryNodeReach/package-lock.json was committed to main with the PR and it should not be there. That is an outdated structure, and as you can see now we just have a folder that only contains a package-lock.

 Just noticed. Correct, it shouldn't be there.

**Comment by prasannarajezzzy** [comment]
> > @sachinprasad1998mav @prasannarajezzzy @murthy-001 @mmendan Please be careful when creating and reviewing PR's. PantryNodeReach/package-lock.json was committed to main with the PR and it should not be there. That is an outdated structure, and as you can see now we just have a folder that only contains a package-lock.
> 
> Just noticed. Correct, it shouldn't be there. Resolved in #203

Yes, we need to add package-lock into the git ignore file please do that

**Comment by hardikpatil** [comment]
It'd also be more interactive if it checks whether the email address that exists in the system or not. or would this feature require a separate API in the backend? @briswells @AbhinavReddy-Dev 

Screenshot 2023-04-27 at 10 38 23

**Comment by hardikpatil** [comment]
> > > @sachinprasad1998mav @prasannarajezzzy @murthy-001 @mmendan Please be careful when creating and reviewing PR's. PantryNodeReach/package-lock.json was committed to main with the PR and it should not be there. That is an outdated structure, and as you can see now we just have a folder that only contains a package-lock.
> > 
> > 
> > Just noticed. Correct, it shouldn't be there. Resolved in #203
> 
> Yes, we need to add package-lock into the git ignore file

nah, it shouldn't be in `.gitignore` 
https://stackoverflow.com/questions/48524417/should-the-package-lock-json-file-be-added-to-gitignore

**Comment by briswells** [comment]
> @sachinprasad1998mav @DevangRaval1 It'd also be more interactive if it checks whether the email address that exists in the system or not. or would this feature require a separate API in the backend? @briswells @AbhinavReddy-Dev
> 
> Screenshot 2023-04-27 at 10 38 23

From what I can see this doesn't touch the backend at all, much like the register feature.

The way this should work is the front end should send the email address to the backend. The backend would than create a code, email it to the user if the email exists. If it doesn't exist it'd do nothing. Regardless of if the email exists or not though the backend will send the same response to the frontend. This stops ppl from using the forget password feature to guess what emails are already in the system. Then the user would input the code from the email, which would then be sent and validated by the backend. After the backend validated it some kind of special authenticated session would be created which was different from normal authenticated session, and only had access to the reset password route. The front end would than send the new password to that backend route which would finally reset the users password. After which the user would need to login using the new password so a properly authenticated session would be created.

As you can see a TON of work is still needed to implement this feature. The Pull that closed this only created the UI for the feature, it did not implement it.

**Comment by sachinprasad1998mav** [comment]
@hardikpatil @briswells As we were in the front-end team, our focus was only on the UI, we have done it as mentioned in the issue. I do accept we should have named the PR better. If anyone else is working on the backend part of forgot password issue, let us know.

**Pull request merged**